setup.py

```diff
@@ -35,14 +35,93 @@ def is_conda_interpreter():
     """
     return 'conda' in sys.version or 'Continuum' in sys.version
 
+def is_macOS():
+    return 'darwin' in sys.platform
+
+
+# HOMEBREW_SAMPLE = """$ brew ls --verbose libomp
+# /opt/homebrew/Cellar/libomp/15.0.7/INSTALL_RECEIPT.json
+# /opt/homebrew/Cellar/libomp/15.0.7/.brew/libomp.rb
+# /opt/homebrew/Cellar/libomp/15.0.7/include/ompt.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp-tools.h
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.dylib
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.a
+# """
+#
+# MACPORTS_SAMPLE = """$ port contents libomp
+# Port libomp contains:
+#   /opt/local/include/libomp/omp-tools.h
+#   /opt/local/include/libomp/omp.h
+#   /opt/local/include/libomp/ompt.h
+#   /opt/local/lib/libomp/libgomp.dylib
+#   /opt/local/lib/libomp/libiomp5.dylib
+#   /opt/local/lib/libomp/libomp.dylib
+#   /opt/local/share/doc/libomp/LICENSE.TXT
+#   /opt/local/share/doc/libomp/README.txt
+# """
+
+def macOS_list_brew_port(cmd):
+    from subprocess import run
+    lib = run(cmd + " | awk '/\/lib\// { print $1; }' | xargs -n1 dirname | uniq",
+              shell=True, check=False, capture_output=True)
+    inc = run(cmd + " | awk '/\/include\// { print $1; }' | xargs -n1 dirname | uniq",
+              shell=True, check=False, capture_output=True)
+    if lib.returncode == 0 and inc.returncode == 0:
+        return inc.stdout.decode("UTF-8").strip(), lib.stdout.decode("UTF-8").strip()
+    return None, None
+
+MACOS_OMP = None
+
+def macOS_libomp_known_locations():
+    """ Find common paths for libomp installation on macOS
+        return flags for compile and link as lists
+        e.g. (['-I/opt/local/include/libomp'], ['-L/opt/local/lib/libomp'])
+    """
+    if not is_macOS():
+        return [], []
+    global MACOS_OMP
+    if MACOS_OMP:
+        return MACOS_OMP
+    for cmd in ["brew ls --verbose libomp", "port contents libomp"]:
+        inc, lib = macOS_list_brew_port(cmd)
+        if inc and lib:
+            MACOS_OMP = [f"-I{inc}"], [f"-L{lib}"]
+            return MACOS_OMP
+    return [], []
+
+def is_macOS_with_libomp():
+    if not is_macOS():
+        return False
+    inc, lib = macOS_libomp_known_locations()
+    return bool(inc) and bool(lib)
+
+
+OMP_TAB = {
+    '1': 'gomp',
+    '0': None,
+    'gcc': 'gomp',
+    'gomp': 'gomp',
+    'clang': 'omp',
+    'omp': 'omp',
```

Comment:
If this main thing this is deciding is gomp versus omp, then shouldn't we be able to determine which one is available through your existing work with `guess`? Meaning we could maybe get rid of `gcc` and `clang` in this dict and the user specifies the compiler to pip/setuptools (ex. `--compiler=gcc`) then we can determine which version of OpenMP should be used? This has the added benefit of people not assuming that `USE_OMP='gcc'` is somehow switching compilers from their default compiler.

If these multiple entries are all a convenience then I can understand that, but it seems like it is distracting from the main purpose of the dictionary which is "do we use gomp or omp". And I think I understand the `TAB` naming, but maybe we get more explicit with the naming if some of these options are removed?

Reply:
In this case, I wanted to retain the possibility of manual installs of libomp, and non-macOS systems using Clang, e.g. 
`CC=/opt/clang/bin/clang USE_OMP=clang CFLAGS=-I/opt/libomp/include LDFLAGS=-L/opt/libomp/lib pip install .`

The default "guess" is intentional, though it could also be called "probe" - especially if other common installation cases come along for clang/libomp . 

I couldn't decide between "clang" or "omp" and likewise "gcc" or "gomp" so I left them both in. Depending on who's at the keyboard, it may be more obvious to use the name of the compiler suite - i.e. it's not necessarily obvious that gomp corresponds to gcc and omp to llvm/clang.

Reply:
I guess that makes sense. We can document `clang` as an alias for `omp` and likewise for `gcc`.